### PR TITLE
HPCC4J-478 Invalidate baremetal specific functionality

### PR DESCRIPTION
**Pull request by rpastrana**
- Rejects invalid containerized requests
- Updates junit tests

Signed-off-by: Rodrigo Pastrana <rodrigo.pastrana@lexisnexisrisk.com>

## Type of change:
- [ ] This change is a bug fix (non-breaking change which fixes an issue).
- [x] This change is a new feature (non-breaking change which adds functionality).
- [ ] This change is a breaking change (fix or feature that will cause existing behavior to change).

## Checklist:
- [x] I have created a corresponding JIRA ticket for this submission
- [x] My code follows the code style of this project.
  - [ ] I have applied the Eclipse code-format template provided.
- [ ] My change requires a change to the documentation.
  - [ ] I have updated the documentation accordingly, or...
  - [ ] I have created a JIRA ticket to update the documentation.
  - [ ] Any new interfaces or exported functions are appropriately commented.
- [x] I have read the HPCC Systems CONTRIBUTORS document (https://github.com/hpcc-systems/HPCC-Platform/wiki/Guide-for-contributors).
- [x] The change has been fully tested:
  - [ ] This change does not cause any existing JUnits to fail.
  - [ ] I have include JUnit coverage to test this change
  - [ ] I have performed system test and covered possible regressions and side effects.
- [x] I have given due consideration to all of the following potential concerns:
  - [ ] Scalability
  - [ ] Performance
  - [ ] Security
  - [ ] Thread-safety
  - [ ] Premature optimization
  - [ ] This change fixes the problem, not just the symptom

### CODE REVIEW

**Comment by rpastrana** [comment]
@jpmcmu please review

File: wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsTopologyClient.java [hunk 1]
```diff
@@ -227,6 +232,100 @@ private void initWsTopologyStub(Connection connection)
...
+
+    /**
+     * Fetch data storage planes
+     *
+     * @param kind enumerated options
+     * @return wrapped response object
+     * @throws HpccContainerizedUnsupportedException if kind not provided while targeting baremetal HPCC environment
+     * @throws Exception if error encountered
+     */
+    public TpGroupQueryResponseWrapper getDataStoragePlanes(TopologyGroupQueryKind kind) throws HpccContainerizedUnsupportedException, Exception
```
Comment (wangkx) [inline wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsTopologyClient.java:283]: Should this getDataStoragePlanes() (and the one below) be renamed to getTopologyGroups()? Does ‘Data Storage Plane’ make sense while targeting a baremetal environment?

Comment (rpastrana) [reply wsclient/src/main/java/org/hpccsystems/ws/client/HPCCWsTopologyClient.java:283]: Good point, now that it is no longer Containerized only function, the name should reflect the new functionality...

**Comment by rpastrana** [comment]
> @jpmcmu please review

bump

**Comment by jpmcmu** [comment]
@rpastrana Looks good, being able to grab the topology group will help clean up some of our tests on bare metal as well.

**Comment by rpastrana** [comment]
@jpmcmu which tests in particular? I can look into including the cleanup in this PR

**Comment by jpmcmu** [comment]
@rpastrana The DFSClient grab a system property to determine the cluster group name in BaseRemoteTest.java. Changing that to grab the topology group if not specified would be great.

**Comment by rpastrana** [comment]
@jpmcmu take a look at latest changes, thanks.

File: wsclient/src/main/java/org/hpccsystems/ws/client/platform/test/BaseRemoteTest.java [hunk 2]
```diff
@@ -199,6 +189,37 @@ public boolean verify(String hostname,javax.net.ssl.SSLSession sslSession)
         try
         {
             wsclient = platform.checkOutHPCCWsClient();
+            if (thorClusterFileGroup == null || thorClusterFileGroup.isEmpty())
+            {
+                List<TpGroupWrapper> grouplist = wsclient.getTopologyGroups(wsclient.isContainerized() ? TopologyGroupQueryKind.PLANE : TopologyGroupQueryKind.THOR);
+                for (TpGroupWrapper tpGroupWrapper : grouplist)
+                {
+                    thorClusterFileGroup = tpGroupWrapper.getName();
+                    if (thorclustername != null)
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/platform/test/BaseRemoteTest.java:198]: Should this be thorClusterFileGroup != null?

File: wsclient/src/main/java/org/hpccsystems/ws/client/platform/test/BaseRemoteTest.java [hunk 3]
```diff
@@ -199,6 +189,37 @@ public boolean verify(String hostname,javax.net.ssl.SSLSession sslSession)
...
+            {
+                System.out.println("RemoteTest: 'thorClusterFileGroup': '" + thorClusterFileGroup + "'");
+            }
+
+            if (roxieClusterGroup == null || roxieClusterGroup.isEmpty())
+            {
+                List<TpGroupWrapper> grouplist =  wsclient.getTopologyGroups(wsclient.isContainerized() ? TopologyGroupQueryKind.PLANE : TopologyGroupQueryKind.ROXIE);
+                for (TpGroupWrapper tpGroupWrapper : grouplist)
+                {
+                	roxieClusterGroup = tpGroupWrapper.getName();
```
Comment (jpmcmu) [inline wsclient/src/main/java/org/hpccsystems/ws/client/platform/test/BaseRemoteTest.java:213]: Formatting

**Review by jpmcmu — COMMENTED** [review]
@rpastrana One potential issue, otherwise looks good

**Review by jpmcmu — APPROVED** [review]
@rpastrana looks good

**Pull request merged**